### PR TITLE
better arguments processing

### PR DESCRIPTION
**Pull request by FabioLuporini**
I think this PR makes understanding and changing the arguments processing **way** easier than in current master

### CODE REVIEW

File: devito/function.py [hunk 1]
```diff
@@ -293,33 +297,35 @@ def _arg_defaults(self, alias=None):
...
-
-        # Add value override for own data if it is provided
+        # Add value override for own data if it is provided, otherwise
+        # use defaults
         if self.name in kwargs:
             new = kwargs.pop(self.name)
             if isinstance(new, TensorFunction):
                 # Set new values and re-derive defaults
-                values.update(new._arg_defaults(alias=key).reduce_all())
+                values = new._arg_defaults(alias=self).reduce_all()
```
Comment (mloubout) [inline devito/function.py]: `alias=self`? shouldn't it stay `key`?

File: devito/function.py [hunk 2]
```diff
@@ -293,33 +297,35 @@ def _arg_defaults(self, alias=None):
...
             else:
                 # We've been provided a pure-data replacement (array)
-                values[key.name] = new
+                values = {self.name: new}
```
Comment (mloubout) [inline devito/function.py]: Same, no more key?

File: devito/function.py (hunk 1 above)
Comment (mloubout) [reply devito/function.py]: ok

**Comment by FabioLuporini** [comment]
Closing and opening a new one to see if jenkins disappears

**Pull request merged**